static/js/favicon.js

```diff
@@ -39,7 +47,16 @@ export function update_favicon(new_message_count, pm_count) {
         }
 
         if (new_message_count === 0 && pm_count === 0) {
-            $("#favicon").attr("href", "/static/images/favicon.svg?v=4");
+            if (user_settings.realm_icon_as_favicon && page_params.realm_icon_url) {
+                // Having both an alternate_icon and the realm_icon set as the
+                // favicon would sometimes only show the alternate_icon, so we
+                // first clear the alternate icon.
+                clear_alternate_icon();
```

Comment:
Can you debug this more? Under what circumstances does this happen? Why does this not happen with our existing dynamic favicon? Or does it?

Reply:
Yeah, I'd also prefer that we debug/reproduce this.

Reply:
> Or does it?

This was key, I think I'm on track to figuring things out. We don't currently notice this behavior because our png and svg icons look the same.

Reply:
huh. or maybe not. I can no longer reproduce any problems if I test with `disable cache` in the network tab checked.